autogalaxy/interferometer/model/analysis.py

```diff
@@ -0,0 +1,190 @@
+from astropy import cosmology as cosmo
+
+import autofit as af
+import autoarray as aa
+
+from autogalaxy.analysis.analysis import AnalysisDataset
+from autogalaxy.interferometer.model.result import ResultInterferometer
+from autogalaxy.analysis.visualizer import Visualizer
+from autogalaxy.interferometer.fit_interferometer import FitInterferometer
+from autogalaxy.galaxy.galaxy import Galaxy
+
+from autogalaxy import exc
+
+
+class AnalysisInterferometer(AnalysisDataset):
+    def __init__(
+        self,
+        dataset,
+        hyper_dataset_result=None,
+        cosmology=cosmo.Planck15,
+        settings_pixelization=aa.pix.SettingsPixelization(),
+        settings_inversion=aa.SettingsInversion(),
+        preloads=aa.Preloads(),
+    ):
+
+        super().__init__(
+            dataset=dataset,
+            hyper_dataset_result=hyper_dataset_result,
+            cosmology=cosmology,
+            settings_pixelization=settings_pixelization,
+            settings_inversion=settings_inversion,
+            preloads=preloads,
+        )
+
+        if self.hyper_dataset_result is not None:
+
+            self.set_hyper_dataset(result=self.hyper_dataset_result)
+
+        else:
+
+            self.hyper_galaxy_visibilities_path_dict = None
+            self.hyper_model_visibilities = None
+
+    def set_hyper_dataset(self, result):
+
+        super().set_hyper_dataset(result=result)
+
+        self.hyper_model_visibilities = result.hyper_model_visibilities

```

Comment:
Again, for long model-fitting pipelines one needs to update the hyper dataset. It would require 5-6 lines of Python to do via making a new Analysis class, so these setters are for a more concise pipeline API.

I agree it pains me to break the "rules" of good code design...